### PR TITLE
Bugfix: QA: Run tests with UPnP disabled

### PR DESCRIPTION
**Pull request by luke-jr**
Needed for builds configured with --enable-upnp-default

### CODE REVIEW

**Comment by promag** [comment]
Not quite a bugfix?

Anyway, without this change:
```sh
./configure  ... --enable-upnp-default
...
checking whether to build with support for UPnP... yes
checking whether to build with UPnP enabled by default... yes
...

make
...
```
And the following fails:
```sh
test/functional/feature_config_args.py
```

Tested ACK be192e6734d4bd59debbd63bb9b135284494d113

**Comment by practicalswift** [comment]
Thanks for finding and addressing this issue @luke-jr -- we should take the security of our users seriously.

ACK be192e6734d4bd59debbd63bb9b135284494d113 -- diff looks correct

@promag If running the tests opens up ports in our users' firewalls (without them explicitly asking for such behaviour) then that definitely sounds like a bug to me :-)

File: test/functional/test_framework/test_node.py [hunk 1]
```diff
@@ -91,6 +91,7 @@ def __init__(self, i, datadir, *, rpchost, timewait, bitcoind, bitcoin_cli, cove
             "-debugexclude=libevent",
             "-debugexclude=leveldb",
             "-uacomment=testnode%d" % i,
+            "-upnp=0",
```
Comment (MarcoFalke) [inline test/functional/test_framework/test_node.py]: See the comment in line 84

`common args are set in the config file (see initialize_datadir)`. The args here are only to set the datadir and debug log options.

**Comment by promag** [comment]
@practicalswift I mean it's not a bug to the end user.

**Comment by laanwj** [comment]
Yes, this seems the correct fix
ACK be192e6734d4bd59debbd63bb9b135284494d113

**Comment by hebasto** [comment]
ACK be192e6734d4bd59debbd63bb9b135284494d113

Verified that after `./configure --enable-upnp-default && make`
the `test/functional/feature_config_args.py` fails on master (e47e36cb4975b332a4d9552bc73b031ab8fd6ab1).

With this PR the `test/functional/feature_config_args.py` passes.

Nit: agree with the suggestion to move `-upnp` option to the `bitcoin.conf`:
https://github.com/bitcoin/bitcoin/blob/e47e36cb4975b332a4d9552bc73b031ab8fd6ab1/test/functional/test_framework/util.py#L289-L305

File: test/functional/test_framework/test_node.py (hunk 1 above)
Comment (promag) [reply test/functional/test_framework/test_node.py]: Agree, looks like all tests use the generated bitcoin.conf.

**Comment by fanquake** [comment]
I've cherry-picked this change into #16646, but modified it to use `bitcoin.conf`.

**Pull request closed without merging**